File: src/main/app/src/NPattern.js
import React, { Component } from 'react';
import PropTypes from 'prop-types';


class Pattern extends Component {
    static propTypes = {
        id: PropTypes.string.isRequired,
        link: PropTypes.string.isRequired,
        size: PropTypes.object
    };

    render() {
        const { id, link, size } = this.props;

        return (
            <defs>
                <pattern id={id} patternUnits="objectBoundingBox" x={0} y={0} width={size.x} height={size.y}>
                    <rect height="100" width="100" fill="white"/>
                    <image xlinkHref={link} x={10} y={10} width={size.x*2} height={size.y*2} />
                </pattern>
            </defs>
        );
    }
}

export default Pattern;

File: src/main/java/de/bonndan/nivio/output/map/NPattern.java
package de.bonndan.nivio.output.map;

import j2html.tags.DomContent;

import static j2html.TagCreator.rawHtml;

class NPattern extends Component {

    private final String id, link;
    private final int size, padding;

    NPattern(String id, String link, int size, int padding) {
        this.id = id;
        this.link = link;
        this.size = size;
        this.padding = padding;
    }


    public DomContent render() {

        return rawHtml(
                "<defs>" +
                        "<pattern id=\"" + id + "\" patternUnits=\"objectBoundingBox\" x =\"0\" y=\"0\" width=\"" + size + "\" height=\"" + size + "\" >" +
                        "<rect height=\"100\" width=\"100\" fill=\"white\" />" +
                        "<image xlink:href=\"" + link + "\" x=\"" + padding + "\" y=\"" + padding + "\" width=\"" + size * 2 + "\" height=\"" + size * 2 + "\"  />" +
                        "</pattern>" +
                        "</defs >"
        );
    }
}

